### PR TITLE
Groups

### PR DESCRIPTION
**Pull request by rlin0**
### Summary
-  Endpoint for get position holders and deleted group_id from position holders

### CODE REVIEW

File: donut/modules/groups/helpers.py [hunk 1]
```diff
@@ -173,6 +173,23 @@ def delete_position(pos_id):
     with flask.g.pymysql_db.cursor() as cursor:
         cursor.execute(s)
 
+# TODO: Implement this helper function
+def create_position_holder(group_id, pos_id, user_id, start_date, end_date):
```
Comment (majesticfish) [inline donut/modules/groups/helpers.py]: Try adding a test for this function

File: donut/modules/groups/routes.py [hunk 2]
```diff
@@ -70,7 +70,30 @@ def get_group_members(group_id):
...
+                validate_exists(form, "user_id")
+                and validate_int(form["user_id"]),
+                validate_exists(form, "start_date")
+                and validate_date(form["start_date"]),
+                validate_exists(form, "end_date")
+                and validate_date(form["end_date"])
+        ]
+        if not all(validations):
+            return jsonify({'success': False})
+        else:
```
Comment (majesticfish) [inline donut/modules/groups/routes.py]: Don't actually need this else

**Review by majesticfish — CHANGES_REQUESTED** [review]
Preferably create a test function to test the create_position_holder method.

File: donut/modules/groups/helpers.py [hunk 3]
```diff
@@ -173,6 +173,23 @@ def delete_position(pos_id):
...
+        group_id: group id of the position
+        pos_id: id of the position
+        user_id: user id of the person that the position is to be assigned
+        start_date: Starting date of the holding period, format is 'yyyy-mm-dd'
+        end_date: end date of the hold period
+    '''
+    s = """INSERT INTO position_holders (pos_id, user_id, start_date, 
+    end_date) VALUES (%s, %s, %s, %s)"""
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(s, (pos_id, user_id, start_date,
```
Comment (majesticfish) [inline donut/modules/groups/helpers.py]: also run 'make lint'

File: donut/modules/groups/helpers.py [hunk 4]
```diff
@@ -173,6 +173,23 @@ def delete_position(pos_id):
     with flask.g.pymysql_db.cursor() as cursor:
         cursor.execute(s)
 
+# TODO: Implement this helper function
```
Comment (calebsander) [inline donut/modules/groups/helpers.py]: Remove `TODO`

File: donut/modules/groups/routes.py [hunk 5]
```diff
@@ -70,7 +70,30 @@ def get_group_members(group_id):
     return jsonify(helpers.get_members_by_group(group_id))
 
 
-@blueprint.route("/1/positions/<int:pos_id>/")
+# TODO: Modify this end point to have both POST and GET methods
```
Comment (calebsander) [inline donut/modules/groups/routes.py]: Remove `TODO`

File: donut/modules/groups/routes.py [hunk 6]
```diff
@@ -70,7 +70,30 @@ def get_group_members(group_id):
...
+# The POST method should use a flask.request.form and call the
+# helper function to create a position holding
+# GET method should just query for position holders
+@blueprint.route("/1/positions/<int:pos_id>/", methods=["POST", "GET"])
```
Comment (calebsander) [inline donut/modules/groups/routes.py]: `POST` and `GET` handlers should go in different functions if they don't share any code:
````python
@blueprint.route("/1/positions/<int:pos_id>/", methods=["GET"])
def get_pos_holders(pos_id):
    return jsonify(helpers.get_position_holders(pos_id))

@blueprint.route("/1/positions/<int:pos_id>/", methods=["POST"])
def add_pos_holder(pos_id):
    # your code...
````

File: donut/modules/groups/routes.py [hunk 7]
```diff
@@ -70,7 +70,30 @@ def get_group_members(group_id):
...
+                and validate_int(form["user_id"]),
+                validate_exists(form, "start_date")
+                and validate_date(form["start_date"]),
+                validate_exists(form, "end_date")
+                and validate_date(form["end_date"])
+        ]
+        if not all(validations):
+            return jsonify({'success': False})
+        else:
+            helpers.create_position_holder(int(pos_id), int(form["user_id"]),
```
Comment (calebsander) [inline donut/modules/groups/routes.py]: `pos_id` should already be an `int`, no?

**Review by calebsander — CHANGES_REQUESTED** [review]
What Sean said, and some small other suggested edits

File: donut/modules/groups/helpers.py [hunk 8]
```diff
@@ -174,6 +174,23 @@ def delete_position(pos_id):
         cursor.execute(s)
 
 
+# TODO: Implement this helper function
```
Comment (calebsander) [inline donut/modules/groups/helpers.py]: Remove `TODO`

**Review by majesticfish — APPROVED** [review]

**Pull request merged**